lua/starfall/libs_sv/ply.lua

```diff
@@ -0,0 +1,211 @@
+return function( instance )
+
+
+local checktype = instance.CheckType
+local player_methods = instance.Types.Player.Methods
+local weapon_methods = instance.Types.Weapon.Methods
+local ply_meta, punwrap = instance.Types.Player, instance.Types.Player.Unwrap
+local ang_meta, awrap, aunwrap = instance.Types.Angle, instance.Types.Angle.Wrap, instance.Types.Angle.Unwrap
+local vec_meta, vwrap, vunwrap = instance.Types.Vector, instance.Types.Vector.Wrap, instance.Types.Vector.Unwrap
+local wep_meta, wwrap, wunwrap = instance.Types.Weapon, instance.Types.Weapon.Wrap, instance.Types.Weapon.Unwrap
+
+
+local function getPly( this )
+    local ent = punwrap( this )
+
+    if ent:IsValid() then
+        return ent
+    else
+        SF.Throw( "Player is not valid.", 3 )
+    end
+end
+
+
+local function canUsePlyCore( play )
+    return play:IsAdmin() or play:IsSuperAdmin()
+end
+
+
+--- Sets the health of a player.
+-- @param number Health
+function player_methods:setHealth( health )
+    if not canUsePlyCore( instance.player ) then SF.Throw( "You must be an admin to use this function!" ) end
+    
+    checktype( self, ply_meta )
+    local ply = getPly( self )
+    ply:SetHealth( health )
+end
+
+
+--- Sets the maximum health of a player. (i.e. How much health a Medkit can heal you to)
+-- @param number MaxHealth
+function player_methods:setMaxHealth( maxHealth )
+    if not canUsePlyCore( instance.player ) then SF.Throw( "You must be an admin to use this function!" ) end
+    
+    checktype( self, ply_meta )
+    local ply = getPly( self )
+    ply:SetMaxHealth( maxHealth )
+end
+
+
+--- Sets the suit armor of a player.
+-- @param number SuitArmor
+function player_methods:setArmor( armor )
```

Comment:
Should we check if `armor` is a number?

Reply:
From my testing, starfall seems to do automatic typechecking, shouldn't be required.

Reply:
I worry it might spam errors in the server console, however.

Reply:
Does that actually happen? When I tested that, it just gave me a starfall error saying there was a type mismatch.